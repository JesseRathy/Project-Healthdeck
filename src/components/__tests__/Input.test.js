import {Input} from '../common/index.js';
import { shallow,configure,mount } from 'enzyme';
import React from 'react';
import configureStore from 'redux-mock-store';
import {connect} from 'react-redux';
import Adapter from 'enzyme-adapter-react-16';
import sinon from "sinon";
//Imports to get Enzyme to work if possible -> will work on later
configure({ adapter: new Adapter() });

test('renders correctly with some random test input', () => {


  const wrapper = shallow(<Input Input= {
    /*I've just created a const of all the values and called the card with these values.
    I have to call it */
  { value:'TestInput' }}/>);
  expect(wrapper).toMatchSnapshot();
});

test('renders correctly with some japanese character test input', () => {


  const wrapper = shallow(<Input Input= {
    /*just call the value out of the input; is the only seemingly important part of the test */
  { value:'ドラゴンツリー' }}/>);
  expect(wrapper).toMatchSnapshot();
});

test('renders correctly with some arabic character test input', () => {


  const wrapper = shallow(<Input Input= {
    /*just call the value out of the input; is the only seemingly important part of the test */
  { value:'البلوط العظيم' }}/>);
  expect(wrapper).toMatchSnapshot();
});

test('renders correctly with some character input with spaces', () => {


  const wrapper = shallow(<Input Input= {
      /*just call the value out of the input; is the only seemingly important part of the test */
  { value:'The Quick Brown Fox Jumped Over the Lazy Dog' }}/>);
  expect(wrapper).toMatchSnapshot();
});

test('renders correctly with some input with hypenated input', () => {


  const wrapper = shallow(<Input Input= {
      /*just call the value out of the input; is the only seemingly important part of the test */
  { value:'Anne-Marie' }}/>);
  expect(wrapper).toMatchSnapshot();
});

test('renders correctly with some input with symbol input', () => {

  const wrapper = shallow(<Input Input= {
      /*just call the value out of the input; is the only seemingly important part of the test */
  { value:'$$@!&*(^)%%' }}/>);
  expect(wrapper).toMatchSnapshot();
});
