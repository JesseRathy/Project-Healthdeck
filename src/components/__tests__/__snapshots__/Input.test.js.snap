// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with some arabic character test input 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Input
    Input={
      Object {
        "value": "البلوط العظيم",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Component
          style={
            Object {
              "flex": 1,
              "fontSize": 18,
              "paddingLeft": 20,
            }
          }
        />,
        <TextInput
          allowFontScaling={true}
          autoCorrect={false}
          style={
            Object {
              "color": "#000",
              "flex": 2,
              "fontSize": 18,
              "lineHeight": 23,
              "paddingLeft": 5,
              "paddingRight": 5,
            }
          }
          underlineColorAndroid="transparent"
        />,
      ],
      "style": Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "height": 40,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": undefined,
          "style": Object {
            "flex": 1,
            "fontSize": 18,
            "paddingLeft": 20,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "allowFontScaling": true,
          "autoCorrect": false,
          "onChangeText": undefined,
          "placeholder": undefined,
          "secureTextEntry": undefined,
          "style": Object {
            "color": "#000",
            "flex": 2,
            "fontSize": 18,
            "lineHeight": 23,
            "paddingLeft": 5,
            "paddingRight": 5,
          },
          "underlineColorAndroid": "transparent",
          "value": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Component
            style={
              Object {
                "flex": 1,
                "fontSize": 18,
                "paddingLeft": 20,
              }
            }
          />,
          <TextInput
            allowFontScaling={true}
            autoCorrect={false}
            style={
              Object {
                "color": "#000",
                "flex": 2,
                "fontSize": 18,
                "lineHeight": 23,
                "paddingLeft": 5,
                "paddingRight": 5,
              }
            }
            underlineColorAndroid="transparent"
          />,
        ],
        "style": Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "height": 40,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": undefined,
            "style": Object {
              "flex": 1,
              "fontSize": 18,
              "paddingLeft": 20,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "allowFontScaling": true,
            "autoCorrect": false,
            "onChangeText": undefined,
            "placeholder": undefined,
            "secureTextEntry": undefined,
            "style": Object {
              "color": "#000",
              "flex": 2,
              "fontSize": 18,
              "lineHeight": 23,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            "underlineColorAndroid": "transparent",
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`renders correctly with some character input with spaces 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Input
    Input={
      Object {
        "value": "The Quick Brown Fox Jumped Over the Lazy Dog",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Component
          style={
            Object {
              "flex": 1,
              "fontSize": 18,
              "paddingLeft": 20,
            }
          }
        />,
        <TextInput
          allowFontScaling={true}
          autoCorrect={false}
          style={
            Object {
              "color": "#000",
              "flex": 2,
              "fontSize": 18,
              "lineHeight": 23,
              "paddingLeft": 5,
              "paddingRight": 5,
            }
          }
          underlineColorAndroid="transparent"
        />,
      ],
      "style": Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "height": 40,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": undefined,
          "style": Object {
            "flex": 1,
            "fontSize": 18,
            "paddingLeft": 20,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "allowFontScaling": true,
          "autoCorrect": false,
          "onChangeText": undefined,
          "placeholder": undefined,
          "secureTextEntry": undefined,
          "style": Object {
            "color": "#000",
            "flex": 2,
            "fontSize": 18,
            "lineHeight": 23,
            "paddingLeft": 5,
            "paddingRight": 5,
          },
          "underlineColorAndroid": "transparent",
          "value": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Component
            style={
              Object {
                "flex": 1,
                "fontSize": 18,
                "paddingLeft": 20,
              }
            }
          />,
          <TextInput
            allowFontScaling={true}
            autoCorrect={false}
            style={
              Object {
                "color": "#000",
                "flex": 2,
                "fontSize": 18,
                "lineHeight": 23,
                "paddingLeft": 5,
                "paddingRight": 5,
              }
            }
            underlineColorAndroid="transparent"
          />,
        ],
        "style": Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "height": 40,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": undefined,
            "style": Object {
              "flex": 1,
              "fontSize": 18,
              "paddingLeft": 20,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "allowFontScaling": true,
            "autoCorrect": false,
            "onChangeText": undefined,
            "placeholder": undefined,
            "secureTextEntry": undefined,
            "style": Object {
              "color": "#000",
              "flex": 2,
              "fontSize": 18,
              "lineHeight": 23,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            "underlineColorAndroid": "transparent",
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`renders correctly with some input with hypenated input 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Input
    Input={
      Object {
        "value": "Anne-Marie",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Component
          style={
            Object {
              "flex": 1,
              "fontSize": 18,
              "paddingLeft": 20,
            }
          }
        />,
        <TextInput
          allowFontScaling={true}
          autoCorrect={false}
          style={
            Object {
              "color": "#000",
              "flex": 2,
              "fontSize": 18,
              "lineHeight": 23,
              "paddingLeft": 5,
              "paddingRight": 5,
            }
          }
          underlineColorAndroid="transparent"
        />,
      ],
      "style": Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "height": 40,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": undefined,
          "style": Object {
            "flex": 1,
            "fontSize": 18,
            "paddingLeft": 20,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "allowFontScaling": true,
          "autoCorrect": false,
          "onChangeText": undefined,
          "placeholder": undefined,
          "secureTextEntry": undefined,
          "style": Object {
            "color": "#000",
            "flex": 2,
            "fontSize": 18,
            "lineHeight": 23,
            "paddingLeft": 5,
            "paddingRight": 5,
          },
          "underlineColorAndroid": "transparent",
          "value": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Component
            style={
              Object {
                "flex": 1,
                "fontSize": 18,
                "paddingLeft": 20,
              }
            }
          />,
          <TextInput
            allowFontScaling={true}
            autoCorrect={false}
            style={
              Object {
                "color": "#000",
                "flex": 2,
                "fontSize": 18,
                "lineHeight": 23,
                "paddingLeft": 5,
                "paddingRight": 5,
              }
            }
            underlineColorAndroid="transparent"
          />,
        ],
        "style": Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "height": 40,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": undefined,
            "style": Object {
              "flex": 1,
              "fontSize": 18,
              "paddingLeft": 20,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "allowFontScaling": true,
            "autoCorrect": false,
            "onChangeText": undefined,
            "placeholder": undefined,
            "secureTextEntry": undefined,
            "style": Object {
              "color": "#000",
              "flex": 2,
              "fontSize": 18,
              "lineHeight": 23,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            "underlineColorAndroid": "transparent",
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`renders correctly with some input with symbol input 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Input
    Input={
      Object {
        "value": "$$@!&*(^)%%",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Component
          style={
            Object {
              "flex": 1,
              "fontSize": 18,
              "paddingLeft": 20,
            }
          }
        />,
        <TextInput
          allowFontScaling={true}
          autoCorrect={false}
          style={
            Object {
              "color": "#000",
              "flex": 2,
              "fontSize": 18,
              "lineHeight": 23,
              "paddingLeft": 5,
              "paddingRight": 5,
            }
          }
          underlineColorAndroid="transparent"
        />,
      ],
      "style": Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "height": 40,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": undefined,
          "style": Object {
            "flex": 1,
            "fontSize": 18,
            "paddingLeft": 20,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "allowFontScaling": true,
          "autoCorrect": false,
          "onChangeText": undefined,
          "placeholder": undefined,
          "secureTextEntry": undefined,
          "style": Object {
            "color": "#000",
            "flex": 2,
            "fontSize": 18,
            "lineHeight": 23,
            "paddingLeft": 5,
            "paddingRight": 5,
          },
          "underlineColorAndroid": "transparent",
          "value": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Component
            style={
              Object {
                "flex": 1,
                "fontSize": 18,
                "paddingLeft": 20,
              }
            }
          />,
          <TextInput
            allowFontScaling={true}
            autoCorrect={false}
            style={
              Object {
                "color": "#000",
                "flex": 2,
                "fontSize": 18,
                "lineHeight": 23,
                "paddingLeft": 5,
                "paddingRight": 5,
              }
            }
            underlineColorAndroid="transparent"
          />,
        ],
        "style": Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "height": 40,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": undefined,
            "style": Object {
              "flex": 1,
              "fontSize": 18,
              "paddingLeft": 20,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "allowFontScaling": true,
            "autoCorrect": false,
            "onChangeText": undefined,
            "placeholder": undefined,
            "secureTextEntry": undefined,
            "style": Object {
              "color": "#000",
              "flex": 2,
              "fontSize": 18,
              "lineHeight": 23,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            "underlineColorAndroid": "transparent",
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`renders correctly with some japanese character test input 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Input
    Input={
      Object {
        "value": "ドラゴンツリー",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Component
          style={
            Object {
              "flex": 1,
              "fontSize": 18,
              "paddingLeft": 20,
            }
          }
        />,
        <TextInput
          allowFontScaling={true}
          autoCorrect={false}
          style={
            Object {
              "color": "#000",
              "flex": 2,
              "fontSize": 18,
              "lineHeight": 23,
              "paddingLeft": 5,
              "paddingRight": 5,
            }
          }
          underlineColorAndroid="transparent"
        />,
      ],
      "style": Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "height": 40,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": undefined,
          "style": Object {
            "flex": 1,
            "fontSize": 18,
            "paddingLeft": 20,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "allowFontScaling": true,
          "autoCorrect": false,
          "onChangeText": undefined,
          "placeholder": undefined,
          "secureTextEntry": undefined,
          "style": Object {
            "color": "#000",
            "flex": 2,
            "fontSize": 18,
            "lineHeight": 23,
            "paddingLeft": 5,
            "paddingRight": 5,
          },
          "underlineColorAndroid": "transparent",
          "value": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Component
            style={
              Object {
                "flex": 1,
                "fontSize": 18,
                "paddingLeft": 20,
              }
            }
          />,
          <TextInput
            allowFontScaling={true}
            autoCorrect={false}
            style={
              Object {
                "color": "#000",
                "flex": 2,
                "fontSize": 18,
                "lineHeight": 23,
                "paddingLeft": 5,
                "paddingRight": 5,
              }
            }
            underlineColorAndroid="transparent"
          />,
        ],
        "style": Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "height": 40,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": undefined,
            "style": Object {
              "flex": 1,
              "fontSize": 18,
              "paddingLeft": 20,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "allowFontScaling": true,
            "autoCorrect": false,
            "onChangeText": undefined,
            "placeholder": undefined,
            "secureTextEntry": undefined,
            "style": Object {
              "color": "#000",
              "flex": 2,
              "fontSize": 18,
              "lineHeight": 23,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            "underlineColorAndroid": "transparent",
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`renders correctly with some random test input 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Input
    Input={
      Object {
        "value": "TestInput",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Component
          style={
            Object {
              "flex": 1,
              "fontSize": 18,
              "paddingLeft": 20,
            }
          }
        />,
        <TextInput
          allowFontScaling={true}
          autoCorrect={false}
          style={
            Object {
              "color": "#000",
              "flex": 2,
              "fontSize": 18,
              "lineHeight": 23,
              "paddingLeft": 5,
              "paddingRight": 5,
            }
          }
          underlineColorAndroid="transparent"
        />,
      ],
      "style": Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "height": 40,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": undefined,
          "style": Object {
            "flex": 1,
            "fontSize": 18,
            "paddingLeft": 20,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "allowFontScaling": true,
          "autoCorrect": false,
          "onChangeText": undefined,
          "placeholder": undefined,
          "secureTextEntry": undefined,
          "style": Object {
            "color": "#000",
            "flex": 2,
            "fontSize": 18,
            "lineHeight": 23,
            "paddingLeft": 5,
            "paddingRight": 5,
          },
          "underlineColorAndroid": "transparent",
          "value": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Component
            style={
              Object {
                "flex": 1,
                "fontSize": 18,
                "paddingLeft": 20,
              }
            }
          />,
          <TextInput
            allowFontScaling={true}
            autoCorrect={false}
            style={
              Object {
                "color": "#000",
                "flex": 2,
                "fontSize": 18,
                "lineHeight": 23,
                "paddingLeft": 5,
                "paddingRight": 5,
              }
            }
            underlineColorAndroid="transparent"
          />,
        ],
        "style": Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "height": 40,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": undefined,
            "style": Object {
              "flex": 1,
              "fontSize": 18,
              "paddingLeft": 20,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "allowFontScaling": true,
            "autoCorrect": false,
            "onChangeText": undefined,
            "placeholder": undefined,
            "secureTextEntry": undefined,
            "style": Object {
              "color": "#000",
              "flex": 2,
              "fontSize": 18,
              "lineHeight": 23,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            "underlineColorAndroid": "transparent",
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
